### PR TITLE
[BB-1246] End sprints with a button and upload spillovers to Google Spreadsheets

### PR DESCRIPTION
**Pull request by Agrendalath**
This adds a uploading spillovers to Google Spreadsheets, ending sprints and activating future ones.
**Note**: Currently this PR's target branch is `bb-1245-vacation` for allowing parallel reviews. I'll switch this to `master` after merging that branch.

## Testing instructions:
1. Set `.env` according to the ticket description.
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Create superuser with `docker-compose -f local.yml run --rm django python manage.py createsuperuser` (the button on frontend should be visible for `staff` users only).
1. Create `frontend/.env.local` and set `REACT_APP_GOOGLE_CLIENT_ID` there.
1. Navigate to `frontend` and run `npm i && npm start`.
1. Log in.
1. Click the red button.
1. Check the spreadsheet for the spillover tickets.

### CODE REVIEW

File: sprints/dashboard/utils.py [hunk 1]
```diff
@@ -89,16 +87,82 @@ def prepare_jql_query(current_sprint: int, future_sprint: int, fields: List[str]
...
+        **prepare_jql_query_active_sprint_tickets(
+            list(issue_fields.values()),
+            settings.SPRINT_STATUS_SPILLOVER,
+        ),
+        maxResults=0,
+    )
+
+
+def prepare_spillover_rows(issues: List[Issue], issue_fields: Dict[str, str]) -> List[List[str]]:
+    """Prepares the Google spreadsheet row in the specified format."""
```
[the remark this answers is not in the document]
Comment (Agrendalath) [reply sprints/dashboard/utils.py]: Sure, done. This is the default format of the spreadsheet:

| Ticket | Status | Sprints | Assignee | Reviewer 1 | Reviewer 2 | Reporter | Story Points | Estimated time | Remaining time | Reason for the spillover |
|--------|--------|---------|----------|------------|------------|----------|--------------|----------------|----------------|--------------------------|

File: sprints/dashboard/tasks.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            for sprint in sprints:
+                if sprint.name.startswith('Sprint'):
+                    future_sprint = sprint
+                    break
+
+        issues: List[Issue] = conn.search_issues(
+            **prepare_jql_query_active_sprint_tickets(
+                list(),  # We don't need any fields here. The `key` attribute will be sufficient.
+                settings.SPRINT_STATUS_SPILLOVER | {settings.SPRINT_STATUS_EXTERNAL_REVIEW,
+                                                    settings.SPRINT_STATUS_RECURRING},
```
[the remark this answers is not in the document]
Comment (Agrendalath) [reply sprints/dashboard/tasks.py]: Ah, definitely. Good catch!

File: sprints/dashboard/libs/google.py [hunk 3]
```diff
@@ -13,21 +13,30 @@
...
-    scopes = ['https://www.googleapis.com/auth/calendar']
+    scopes = [
+        'https://www.googleapis.com/auth/calendar',
+        'https://www.googleapis.com/auth/spreadsheets',
+    ]
     credentials = service_account.Credentials.from_service_account_info(GOOGLE_API_CREDENTIALS, scopes=scopes)
-    service = discovery.build('calendar', 'v3', credentials=credentials)
+    if service_name == 'calendar':
+        service = discovery.build(service_name, 'v3', credentials=credentials, cache_discovery=False)
+    elif service_name == 'sheets':
```
[the remark this answers is not in the document]
Comment (Agrendalath) [reply sprints/dashboard/libs/google.py]: Neat!

**Comment by viadanna** [comment]
@Agrendalath I'm still stuck at testing this, specifically the Google Sheets integration. I've set the app id but I'm still getting a 401 status. Everything else works fine and the code looks good to me as well, so this is the only thing blocking the merge. Any ideas?

**Comment by Agrendalath** [comment]
@viadanna, could you please post the related logs on the ticket? I just double-checked everything on a fresh environment and it's working correctly for me.

**Pull request merged**